File: modules/common-net/src/main/java/com/shterneregen/securelan/common/net/transport/ServerSocketFactory.java
package com.shterneregen.securelan.common.net.transport;

import java.io.IOException;
import java.net.ServerSocket;

@FunctionalInterface
public interface ServerSocketFactory {
    ServerSocket open(int port) throws IOException;

    static ServerSocketFactory systemDefault() {
        return ServerSocket::new;
    }
}

File: modules/common-net/src/main/kotlin/com/shterneregen/securelan/common/net/transport/ServerSocketFactory.kt
package com.shterneregen.securelan.common.net.transport

import java.io.IOException
import java.net.ServerSocket

fun interface ServerSocketFactory {
    @Throws(IOException::class)
    fun open(port: Int): ServerSocket

    companion object {
        @JvmStatic
        fun systemDefault(): ServerSocketFactory = ServerSocketFactory { port -> ServerSocket(port) }
    }
}
